### PR TITLE
[DPE-5829] feat: Switch to 24.04 base

### PR DESCRIPTION
**Pull request by zmraul**
- Update base to 24.04
- Switch to `_daemon_` user for the main process. [Reasoning](https://discourse.ubuntu.com/t/unifying-user-identity-across-snaps-and-rocks/36469)

#### **TODO**
- [x] Change snap channel to the correct one one snap is merged:  `stage-snaps: - opensearch/2/edge`

### CODE REVIEW

File: rockcraft.yaml [hunk 1]
```diff
@@ -46,51 +48,47 @@ parts:
...
       cp snap.opensearch/snapcraft.yaml ${rocks}
-      
 
   non-root-user:
     plugin: nil
     after: [opensearch-snap]
-    overlay-script: |
-      # Create a user in the $CRAFT_OVERLAY chroot
-      groupadd -R ${CRAFT_OVERLAY} -g 1000 opensearch
-      useradd -R ${CRAFT_OVERLAY} -M -r -g opensearch -u 1000 opensearch
```
Comment (Mehdi-Bendriss) [inline rockcraft.yaml:80]: why are we deleting the `opensearch` user?

File: scripts/start.sh [hunk 2]
```diff
@@ -88,8 +88,4 @@ sed -i "s@=logs/@=${OPENSEARCH_VARLOG}/@" "${OPENSEARCH_PATH_CONF}/jvm.options"
 
 cat "${conf}"
 
-exec /usr/bin/setpriv \
-  --clear-groups \
-  --reuid opensearch \
-  --regid opensearch \
-  -- "${OPENSEARCH_BIN}"/opensearch
+exec "${OPENSEARCH_BIN}"/opensearch
```
Comment (Mehdi-Bendriss) [inline scripts/start.sh:91]: This will effectively runs as root, `setpriv` ensured that the opensearch service is started with the `opensearch` user.

**Review by Mehdi-Bendriss — COMMENTED** [review]
Thanks Raul - I have questions on the changes related to the opensearch user

File: rockcraft.yaml (hunk 1 above)
Comment (zmraul) [reply rockcraft.yaml:80]: https://discourse.ubuntu.com/t/unifying-user-identity-across-snaps-and-rocks/36469

Adding `run_user: _daemon_` will take care of the start process always being a non-root user.

File: scripts/start.sh (hunk 2 above)
Comment (zmraul) [reply scripts/start.sh:91]: Same as previous comment. User now is `_daemon_` across the board, so no need to fiddle with permissions here. This is inside de container when running:
```
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND                                                                                                   
_daemon_       1  0.0  0.1 1231852 2808 pts/0    Ssl+ 10:30   0:00 /usr/bin/pebble enter                                                                                     
_daemon_      12 71.9 67.8 3268000 1281020 pts/0 Sl   10:30   0:11 /usr/lib/jvm/java-21-openjdk-amd64/bin/java -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopense
```

**Review by Mehdi-Bendriss — DISMISSED** [review]

File: rockcraft.yaml [hunk 3]
```diff
@@ -46,51 +48,47 @@ parts:
   opensearch-snap:
     plugin: nil
     stage-snaps:
-      - opensearch/2/edge
+      - opensearch/2/edge/core2404
```
Comment (phvalguima) [inline rockcraft.yaml]: This snap does not exist. AFAIU this is a "TODO" to be removed once the snap is correctly finished and moves to core24.

**Review by phvalguima — DISMISSED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]

**Pull request merged**